mkdocs/docs/backends/gpu-vulkan.md

```diff
@@ -1,10 +1,173 @@
-# Vulkan GPU backend
+# Vulkan GPU HAL Driver
```

Comment:
Another bikeshedding detail: I'm trying to be careful with terminology.

We can only define a certain number of project-specific terms before people get lost. That's particular important for section headings and titles. While "Vulkan GPU HAL Driver" is more precise, I felt that "Vulkan GPU backend" mapped more directly to the high level "frontends/backends/bindings" categories. Once people are more familiar with the way IREE is structured, they can understand that the backends are implemented via HAL (Hardware Abstraction Layer) drivers.

See also how I'm using https://squidfunk.github.io/mkdocs-material/reference/abbreviations/ on other pages:

![image](https://user-images.githubusercontent.com/4010439/115590244-9808dd80-a285-11eb-8554-1efb3c2be29a.png)

Reply:
That's where I'm coming from here, but again, none of this is set in stone. I could see repharsing the "Backends" section to "Runtime backends" or "Runtime HAL drivers", for example.

Reply:
IMO, the frontend/backend terms are more of how compilers see the world. And as relative (front vs back) terms they can be misleading: what is considered as front/back? Is TF more front than XLA? Can XLA be considered as back comparing to TF? ;-P Also for backend, it can also mean compiler target/backends. So if just seeing IREE backends, it can mean either the IREE HAL driver or IREE compiler target backends. It's overloaded there.

I typically prefer terms that directly defining themselves (so it's not relative and relying on understanding other terms). In ML domain, it might make sense to use (model authoring) framework, (execution) runtime maybe? When we talk about the traditional CodeGen part (which involves just the kernel compilation) we can use compiler terms like backend/target/etc as those are well understood. This avoids us using all compiler terms for everything to avoid overloaded terms. 

WDYT? :)

Reply:
I like "frameworks", and good point about backend/target being compiler terms... brainstorming more section names:

Frontends
Integrations
Frameworks
Framework integrations
Framework importers

Execution runtimes
Executors
Runtime drivers
Runtime HAL drivers
Runtime backends

Bindings
APIs

---

My main gripe with "HAL driver" on its own is that users may not be familiar with the Hardware Abstraction Layer meaning of HAL or make the connection to "runtime" vs "compiler".

Reply:
I'd vote for "executor". It's concise and clear from the word itself. It's also used in ML field already.

Reply:
I'll land this as-is for now; we can do a pass to convert everywhere afterwards. :)

Reply:
I'm less a fan of "executor" _because_ of existing uses and how we don't already use that term anywhere in IREE. The runtime / delegate / HAL driver / etc. distinctions can be subtle for new users.

Anyway, let's follow-up separately (maybe discuss on Discord or a GitHub issue)